types/logger.d.ts

```diff
@@ -60,7 +60,7 @@ export interface FastifyLoggerOptions<
       [key: string]: unknown;
     };
     res?: (res: RawReply) => {
-      statusCode: string | number;
+      statusCode?: string | number;
```

Comment:
It does not seem the test above verifies this is optional.

Reply:
The `test` does reflect it is optional since `statusCode` is not return inside `res`.
Not a direct assertion but just same outcome.